code/processes/gateway.q

```diff
@@ -450,7 +450,8 @@ syncexecjt:{[query;servertype;joinfunction;timeout]
  // doesn't make sense to allow specification of a callback for sync requests
  asyncexecjpts[query;servertype;joinfunction;();timeout;1b];
  // defer response
- @[-30!;(::);()];
+ e:@[{-30!x;1b};(::);0b];
```

Comment:
if this fails and we've already called `asyncexecjpts` above, does that mean query is already in `.gw.queryqueue`? I think possibly that'll mean the query will be run even though the result won't be sent.

Maybe better to move the call to `asyncexecjpts` until after this & only do that if the defer has succeeded?

Reply:
would you add this as another if like if[e..] or change the original if to a $ conditional, and keep the [] you wanted to remove for clarity?

Reply:
have moved this to after now